similar/main/state.cpp

```diff
@@ -1431,10 +1431,19 @@ int state_restore_all_sub(const char *filename, const secret_restore secret)
 // Read the mission info...
 	PHYSFS_read(fp, mission, sizeof(char) * 9, 1);
 
-	if (!load_mission_by_name( mission ))	{
+	if (!load_mission_by_name( mission )
```

Comment:
This will attempt to `load_mission_by_name` on `new_miss` before it notices that this is a special case.  This could lead to surprising results if someone has a loadable mission with that name.  Is that intended?